azure-webapp-maven-plugin/src/main/java/com/microsoft/azure/maven/webapp/ConfigMojo.java

```diff
@@ -372,10 +417,177 @@ private String getDefaultValue(String defaultValue, String fallBack, String patt
     }
 
     private boolean isJarProject() {
-        return getProject().getPackaging().equalsIgnoreCase("jar");
+        return Utils.isJarPackagingProject(project.getPackaging());
     }
 
     private WebAppConfiguration getWebAppConfigurationWithoutValidation() throws AzureExecutionException {
         return new V2NoValidationConfigurationParser(this, new V2ConfigurationValidator(this)).getWebAppConfiguration();
     }
+
+    private WebAppConfiguration chooseExistingWebappForConfiguration()
+            throws AzureExecutionException, AzureAuthFailureException {
+        final Azure az = getAzureClientByAuthType();
+        final TextIO textIO = TextIoFactory.getTextIO();
+        final Subscription[] subscriptions = az.subscriptions().list().toArray(new Subscription[0]);
```

Comment:
why convert it into array?

Reply:
It is better to consolidate the subs list, since  az.subscriptions().list() will return a PagedList, which performs strangely other than common ArrayList